### DISCUSSION TITLE
How to keep references window open?

### DISCUSSION THREAD

**Discussion by talha131** [neoclide/coc.nvim#4351] [Q&A] [answered]
I have enabled this setting from the README.md

```
nmap <silent> gr <Plug>(coc-references)
```

I want to keep the references list window open. But it automatically closes when I select a "use" from the list.

**Comment by chemzqm** [comment, marked as answer]
It's expected to auto close on default action, you can use `CocListResume` to reopen it, but it's recommended to use `<Plug>(coc-refactor)` or use multiple selection feature to open all selected items.

**Reply by talha131** [reply]
Thank you @chemzqm 

All three options that you suggested are good enough to solve my issue.

1. Use `<Plug>(coc-refactor)` like `nmap <leader>cv  <Plug>(coc-refactor)`
2. Use `CocListResume`
3. Do multiple selection and open all the selected references

**Comment by chemzqm** [comment]
See `:h CocLocationsChange` for change options of locations list, or you can use  other UI to display them.

**Comment by talha131** [comment]
One more question please.

References feature use `CocList`, right? If that's the case then shouldn't we have its configurations under `list.source` like other sources. 

For example, we can configure `list.source.files` with the following line in `CocConfig`.

```
  "list.source.files.args": ["--hidden", "--exclude", ".git/"],
```

So a similar setting but for references.

**Reply by chemzqm** [reply]
There are `list.source.location.defaultArgs` and `list.source.location.defaultOptions` but you need to configure `CocLocationsChange` autocmd

**Reply by talha131** [reply]
Thanks a lot for your time and patience.
